gson/src/test/java/com/google/gson/functional/CustomTypeAdaptersTest.java

```diff
@@ -125,7 +125,8 @@ public void testCustomNestedSerializers() {
       }
     }).create();
     ClassWithCustomTypeConverter target = new ClassWithCustomTypeConverter();
-    assertThat(gson.toJson(target)).isEqualTo("{\"bag\":6,\"value\":10}");
+    String targetStr = gson.toJson(target);
+    assertThat(targetStr.equals("{\"bag\":6,\"value\":10}") || targetStr.equals("{\"value\":10,\"bag\":6}")).isTrue();
```

Comment:
Here, instead of using `.equals()`s, you can you the `.isAnyOf(obj1, obj2)` method.
According with the `.isAnyOf()` [documentation](https://truth.dev/api/latest/com/google/common/truth/Subject.html#isAnyOf(java.lang.Object,java.lang.Object,java.lang.Object...)): 
> Fails unless the subject is equal to any of the given elements.

The assert could be something like:
```suggestion
    assertThat(gson.toJson(target)).isAnyOf("{\"bag\":6,\"value\":10}", "{\"value\":10,\"bag\":6}");
```

---
 
Note: I am not a maintainer of the project, please wait for a maintainer's review